component/client/chat/ClientIndependentChatRoom.tsx

```diff
@@ -22,27 +23,24 @@ class ClientIndependentChatRoom extends React.Component<any, IClientIndependentC
     super();
     this.state = {
       currentUrl: Constants.NOOP_URL,
+      userID: Constants.NOOP_ID,
       username: Constants.NOOP_USERNAME
     };
-    storage.get("username")
-      .then((username) => {
-        if (username) {
+    storage.get(Constants.STORAGE_KEY_USERNAME).then((username) => {
+      storage.get(Constants.STORAGE_KEY_ID).then((userID) => {
```

Comment:
> The key idea is "if these calls are very close". The reason for this optimization is that some functionalities tend to break because we are not getting these "very close" React optimizations. This occasionally throws errors in our console and are hard to reproduce because no one has any idea of why it is occurring. This is the stall rendering of a component until we can at least guarantee that it has all the required data it needs to render.